client-app/core/composables/useValidationErrorTranslator.ts

```diff
@@ -0,0 +1,11 @@
+import { useI18n } from "vue-i18n";
+import type { ValidationErrorType } from "@/xapi/types";
+
+export function useValidationErrorTranslator() {
+  const { t, te } = useI18n();
+
+  return (error: ValidationErrorType): string | undefined => {
+    const translationKey = "validation_error." + error.errorCode;
+    return te(translationKey) ? t(translationKey) : error.errorMessage;
```

Comment:
Pass on error parameters `error.errorParameters` to `t(translationKey, params)`
As: `{ "<key1>": "value1", "<key2>": "value2", ... }`